manifest.json

```diff
@@ -1,5 +1,809 @@
+                                }
+                              }
+                            },
+                            "wordValue": {
+                              "default": 0.1,
+                              "type": "number"
+                            }
+                          },
+                          "required": [
+                            "html",
+                            "wordValue"
+                          ]
+                        }
+                      },
+                      "required": [
+                        "role",
+                        "multiplier",
+                        "rewards"
+                      ]
+                    }
+                  },
+                  "wordCountExponent": {
+                    "default": 0.85,
+                    "type": "number"
+                  }
+                },
+                "required": [
+                  "multipliers",
+                  "wordCountExponent"
+                ]
+              },
+              {
+                "type": "null"
+              }
+            ]
+          },
+          "permitGeneration": {
+            "default": null,
+            "anyOf": [
+              {
+                "type": "object",
+                "properties": {}
+              },
+              {
+                "type": "null"
+              }
+            ]
+          },
+          "githubComment": {
+            "default": null,
+            "anyOf": [
+              {
+                "default": {},
+                "type": "object",
+                "properties": {
+                  "post": {
+                    "default": true,
+                    "type": "boolean"
+                  },
+                  "debug": {
+                    "default": false,
+                    "type": "boolean"
+                  }
+                },
+                "required": [
+                  "post",
+                  "debug"
+                ]
+              },
+              {
+                "type": "null"
+              }
+            ]
+          }
+        },
+        "required": [
+          "requirePriceLabel",
+          "contentEvaluator",
+          "userExtractor",
+          "dataPurge",
+          "formattingEvaluator",
+          "permitGeneration",
+          "githubComment"
+        ]
+      },
+      "dataCollection": {
+        "default": {},
+        "type": "object",
+        "properties": {
+          "maxAttempts": {
+            "default": 10,
+            "minimum": 1,
+            "type": "number"
+          },
+          "delayMs": {
+            "default": 1000,
+            "minimum": 100,
+            "type": "number"
+          }
+        },
+        "required": [
+          "maxAttempts",
+          "delayMs"
+        ]
+      }
+    },
+    "required": [
+      "logLevel",
+      "evmNetworkId",
+      "evmPrivateEncrypted",
+      "erc20RewardToken",
+      "incentives",
+      "dataCollection"
+    ]
+  }
+}
```

Comment:
I still have a gut feeling that this should be in a separate file. One is meant to be hand edited by the plugin developer, and another is automatically generated and should not be manually edited by any developer. Same as how other tools do it.

Reply:
The main advantage of having it all in one file is to reduce the amount of network calls made by the kernel because the limit on free tier is very low. And we could aim to fully auto-generate the manifest because all the information can be deduced from the code, which would be nice.